From d5edab7701eb171422f12b7f4624c26c4e569177 Mon Sep 17 00:00:00 2001
From: Bing Xu <bing.xu@intel.com>
Date: Wed, 26 Apr 2023 22:17:35 +0800
Subject: [PATCH] [WA] Set all packages to not killable

App will be killed and not displayed on launcher by carwatchdogd when
the app use more I/O operations, we can disable the feature
"Ensure Peak Performance" on App Settings to avoid this, but the value
can't be saved as Google limitation, Google has fixed the issue on
android 12.1.0.r27, so we disable the feature now.

Tracked-On: OAM-108706
Signed-off-by: Bing Xu <bing.xu@intel.com>
---
 .../src/com/android/car/watchdog/WatchdogPerfHandler.java  | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/service/src/com/android/car/watchdog/WatchdogPerfHandler.java b/service/src/com/android/car/watchdog/WatchdogPerfHandler.java
index 8d15e0f2c..b8f4f48c9 100644
--- a/service/src/com/android/car/watchdog/WatchdogPerfHandler.java
+++ b/service/src/com/android/car/watchdog/WatchdogPerfHandler.java
@@ -72,6 +72,7 @@ import android.os.UserManager;
 import android.util.ArrayMap;
 import android.util.ArraySet;
 import android.util.IndentingPrintWriter;
+import android.util.Log;
 import android.util.SparseArray;
 
 import com.android.internal.annotations.GuardedBy;
@@ -414,9 +415,7 @@ public final class WatchdogPerfHandler {
                 String key = getUserPackageUniqueId(userId, packageInfo.packageName);
                 PackageResourceUsage usage = mUsageByUserPackage.getOrDefault(key,
                         new PackageResourceUsage(userId, packageInfo.packageName));
-                int killableState = usage.syncAndFetchKillableStateLocked(
-                        mPackageInfoHandler.getComponentType(packageInfo.packageName,
-                                packageInfo.applicationInfo));
+                int killableState = KILLABLE_STATE_NEVER;
                 mUsageByUserPackage.put(key, usage);
                 states.add(
                         new PackageKillableState(packageInfo.packageName, userId, killableState));
@@ -1194,7 +1193,7 @@ public final class WatchdogPerfHandler {
         }
 
         public @KillableState int getKillableState() {
-            return mKillableState;
+            return KILLABLE_STATE_NEVER;
         }
 
         public boolean setKillableState(boolean isKillable) {
-- 
2.34.1

